core/core/validators.py

```diff
@@ -2,48 +2,41 @@
 from datetime import datetime
 
 from core.database import Database
-from core.misc import (
-    InvalidBirthdayError,
-    InvalidEmailError,
-    InvalidNameError,
-    InvalidPhoneError,
-    InvalidPhoneLengthError,
-)
 
+class Validator:
+    def validate_existing_contact(name):
+        database = Database()
+        record = database[name]
 
-def validate_existing_contact(name):
-    database = Database()
-    record = database[name]
+        if record:
+            return record
+        else:
+            raise KeyError
 
-    if record:
-        return record
-    else:
-        raise KeyError
 
+    def validate_phone_number(phone_number):
+        phone_number_pattern = re.compile(r"^[0-9+-]+$")
 
-def validate_phone_number(phone_number):
-    phone_number_pattern = re.compile(r"^[0-9+-]+$")
+        if phone_number and len(phone_number) != 10:
+            return True
+        elif phone_number and not phone_number_pattern.match(phone_number):
+            return True
 
-    if len(phone_number) != 10:
-        raise InvalidPhoneLengthError()
-    elif not phone_number_pattern.match(phone_number):
-        raise InvalidPhoneError()
 
+    def validate_name(name):
+        if len(name) == 0 or name is None:
+            return True
```

Comment:
1. You check len and then if name is None. It should be in reverse way. I am not sure if lan(None) will work correctly, so for safety better to change it positions
2. It can be done in one line
3. It looks like it works in opposite way, if name is valid it will return False

`return name and len(name.strip())`